revoke/revoke.go

```diff
@@ -22,14 +24,139 @@ import (
 	"github.com/cloudflare/cfssl/log"
 )
 
-// HardFail determines whether the failure to check the revocation
-// status of a certificate (i.e. due to network failure) causes
-// verification to fail (a hard failure).
-var HardFail = false
+type localCRL struct {
+	// embedded Mutex locker for the localCRL
+	sync.RWMutex
+	path string
+	crl  *pkix.CertificateList
+}
+
+type crlSet struct {
+	// embedded Mutex locker for the crlSet
+	sync.RWMutex
+	// crlSet associates a PKIX certificate list with the URL the CRL is
+	// fetched from.
+	set map[string]*pkix.CertificateList
+}
+
+type hardFail struct {
+	// embedded Mutex locker for the hardFail
+	sync.RWMutex
+	// hardFail value
+	val bool
+}
+
+// Revoke type contains configuration for each new revoke instance
+type Revoke struct {
+	// LocalCRL contains path to the local CRL file. When set, certificate
+	// will be checked only using local CRL script, remote methods will be
+	// skipped.
+	localCRL localCRL
+	// HardFail determines whether the failure to check the revocation
+	// status of a certificate (i.e. due to network failure) causes
+	// verification to fail (a hard failure).
+	hardFail hardFail
+	// crlSet associates a PKIX certificate list with the URL the CRL is
+	// fetched from.
+	crl crlSet
+}
+
+// defaultChecker is a default config for regular apps which don't need to
+// use custom options.
+var defaultChecker = New(false)
+
+// New creates Revoke config structure.
+// Accepts hardfail bool variable as an option
+func New(hardfail bool) *Revoke {
+	return &Revoke{
+		localCRL: localCRL{
+			path: "",
+			crl:  nil,
+		},
+		hardFail: hardFail{
+			val: hardfail,
+		},
+		crl: crlSet{
+			set: map[string]*pkix.CertificateList{},
+		},
+	}
+}
+
+func (r *Revoke) unsetLocalCRL() {
+	r.localCRL.path = ""
+	r.localCRL.crl = nil
+}
+
+// SetLocalCRL sets localCRL path into the Revoke struct
+func SetLocalCRL(localCRLpath string) error {
+	return defaultChecker.SetLocalCRL(localCRLpath)
+}
 
-// CRLSet associates a PKIX certificate list with the URL the CRL is
-// fetched from.
-var CRLSet = map[string]*pkix.CertificateList{}
+// SetLocalCRL sets localCRL path into the Revoke struct
+func (r *Revoke) SetLocalCRL(localCRLpath string) error {
+	r.localCRL.Lock()
+	defer r.localCRL.Unlock()
+
+	return r.setLocalCRL(localCRLpath)
+}
+
+func (r *Revoke) setLocalCRL(localCRLpath string) error {
+	if localCRLpath == "" {
+		r.unsetLocalCRL()
+		return nil
+	}
+
+	if u, err := neturl.Parse(localCRLpath); err != nil {
+		r.unsetLocalCRL()
+		return err
+	} else if u.Scheme == "" {
```

Comment:
style issue:
don't do if err != nil {} else if {}, "if err != nil" should be standalone, especially when error will be returned immediately.  It makes code logic hard to follow.